File: rb/features.rb
# InvoiceExtraction SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module InvoiceExtractionFeatures
  def self.make_feature(name)
    case name
    when "base"
      InvoiceExtractionBaseFeature.new
    when "test"
      InvoiceExtractionTestFeature.new
    else
      InvoiceExtractionBaseFeature.new
    end
  end
end

File: py/invoiceextraction_sdk/features.py
# InvoiceExtraction SDK feature factory

from invoiceextraction_sdk.feature.base_feature import InvoiceExtractionBaseFeature
from invoiceextraction_sdk.feature.test_feature import InvoiceExtractionTestFeature


def _make_feature(name):
    features = {
        "base": lambda: InvoiceExtractionBaseFeature(),
        "test": lambda: InvoiceExtractionTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
